jupyter-docker/Dockerfile

```diff
@@ -76,32 +70,29 @@ RUN export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" \
  && apt-get install -yq --no-install-recommends \
     google-cloud-sdk \
  && apt-get clean \
- && rm -rf /var/lib/apt/lists/* \
+ && rm -rf /var/lib/apt/lists/*
 
- # Java
- && wget --header "Cookie: oraclelicense=accept-securebackup-cookie" $JAVA_URL \
+# Java
+RUN wget --header "Cookie: oraclelicense=accept-securebackup-cookie" $JAVA_URL \
  && mkdir -p /usr/lib/jdk && tar -zxf $JAVA_TGZ -C /usr/lib/jdk \
  && update-alternatives --install /usr/bin/java java $JAVA_HOME/bin/java 100 \
  && update-alternatives --install /usr/bin/javac javac $JAVA_HOME/bin/javac 100 \
- && rm $JAVA_TGZ \
-
- # Spark
- && wget $SPARK_URL && mkdir /spark && tar -zxf $SPARK_TGZ -C /spark \
- && cd $SPARK_HOME && ./build/mvn -Pyarn -DskipTests clean package && cd - \
- && rm $SPARK_TGZ \
- # Needed to access gs:// files in Spark/Hail
- && wget -q https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-latest-hadoop2.jar -O $SPARK_HOME/gcs-connector.jar \
+ && rm $JAVA_TGZ
 
- # Hail
+# Spark / Hadoop / Hail
+# Make empty conf dirs for the update-alternatives commands; they will be mounted from the Dataproc VM
+RUN mkdir -p /etc/spark/conf.dist && mkdir -p /etc/hadoop/conf.empty \
+ && update-alternatives --install /etc/spark/conf spark-conf /etc/spark/conf.dist 100 \
+ && update-alternatives --install /etc/hadoop/conf hadoop-conf /etc/hadoop/conf.empty 100 \
  && mkdir $HAIL_HOME && cd $HAIL_HOME \
  && wget http://storage.googleapis.com/hail-common/builds/0.1/jars/$HAILJAR \
  && wget http://storage.googleapis.com/hail-common/builds/0.1/python/$HAILPYTHON \
  && wget http://storage.googleapis.com/hail-common/distributions/0.1/$HAILZIP \
- && cd - \
-
- # NOTE! not sure why, but this must run before pip installation
- && useradd -m -s /bin/bash -N -u $UID $USER \
+ && cd -
 
+# Pips
+# NOTE! not sure why, but this must run before pip installation
+RUN useradd -m -s /bin/bash -N -u $UID $USER \
  # ubuntu's default pip doesn't work well with jupyter
```

Comment:
I know you didn't write this, but ubuntu? this is jessie...

Reply:
Whoops